lib/src/test/kotlin/org/didcommx/didcomm/pack/encrypted/PackAuthEncryptedTest.kt

```diff
@@ -0,0 +1,194 @@
+import kotlin.test.assertFalse
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+data class PackAuthEncryptedTestData(
+    val msg: Message,
+    val authAlg: AuthCryptAlg,
+    val anonAlg: AnonCryptAlg,
+    var curveType: KeyAgreementCurveType,
+    val signedFrom: String,
+    val protectSenderId: Boolean,
+    var to: String,
+    var from: String
+)
+
+class PackAuthEncryptedTest {
+    companion object {
+
+        @JvmStatic
+        fun packAuthEncryptedTest(): Stream<PackAuthEncryptedTestData> {
+            val toList =
+                getKeyAgreementMethodsInSecrets(Person.BOB, KeyAgreementCurveType.X25519).map { it.id }.toMutableList()
+            toList.add(JWM.BOB_DID)
+
+            val fromList = getKeyAgreementMethodsInSecrets(Person.ALICE).map { it.id }.toMutableList()
+            fromList.add(JWM.ALICE_DID)
+
+            val signedFromList = getAuthMethodsInSecrets(Person.ALICE).map { it.id }.toMutableList()
+            signedFromList.add(JWM.ALICE_DID)
+
+            val cartesianProduct = cartesianProduct(
+                listOf(JWM.PLAINTEXT_MESSAGE, attachmentMulti1msg(), attachmentJsonMsg()),
+                listOf(
+                    AuthCryptAlg.A256CBC_HS512_ECDH_1PU_A256KW
+                ),
+                listOf(
+                    AnonCryptAlg.XC20P_ECDH_ES_A256KW,
+                    AnonCryptAlg.A256GCM_ECDH_ES_A256KW,
+                    AnonCryptAlg.A256CBC_HS512_ECDH_ES_A256KW
+                ),
+                listOf(
+                    KeyAgreementCurveType.X25519,
+                    KeyAgreementCurveType.P256,
+                    KeyAgreementCurveType.P521,
+                    KeyAgreementCurveType.P384
+                ),
+                signedFromList,
+                listOf(true, false),
+                toList,
+                fromList
+            )
+            var stream = Stream.of<PackAuthEncryptedTestData>()
+
+            for (i in cartesianProduct.indices) {
+                stream = Stream.concat(
+                    stream,
+                    Stream.of(
+                        PackAuthEncryptedTestData(
+                            cartesianProduct[i][0] as Message,
+                            cartesianProduct[i][1] as AuthCryptAlg,
+                            cartesianProduct[i][2] as AnonCryptAlg,
+                            cartesianProduct[i][3] as KeyAgreementCurveType,
+                            cartesianProduct[i][4] as String,
+                            cartesianProduct[i][5] as Boolean,
+                            cartesianProduct[i][6] as String,
+                            cartesianProduct[i][7] as String
+                        )
+                    )
+                )
+            }
+
+            return stream
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("packAuthEncryptedTest")
+    fun testAuthcryptSenderDIDRecipientDID(data: PackAuthEncryptedTestData) {
+        data.to = JWM.BOB_DID
```

Comment:
Why do we replace the to/from as prepared in the parameters (fromList, toList)? Can we just pass all parameters (combinations) into one test method? This will in particular test SenderDIDRecipientDID and SenderDIDRecipientKid cases.